.openshift-ci/sensor_versions.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+"""
+Returns the latest patches of the last 4 major versions
+"""
+
+import sys
+import re
+import subprocess
+
+def isReleaseTagGitCLI(version):
+    return bool(re.search(r"^\d+\.\d+\.\d+$", version))
+
+def filterGitCLITags(rawtags):
+    filteredtags = []
+    for t in rawtags:
+        if isReleaseTagGitCLI(t):
+            filteredtags.append(t)
+    return set(filteredtags)
+
+def cliOutputToTags(stdoutput):
+    separated = stdoutput.decode(encoding="utf-8").splitlines()
+    return separated
+
+def transformTagsToNumbers(tags):
+    numTags = []
+    for t in tags:
+        numTags.append(splitVersion(t))
+    return sorted(numTags)
+
+def splitVersion(version):
+    digits = re.search(r"(\d+)\.(\d+)\.\D*(\d+)", version)
+    return int(digits.group(1)), int(digits.group(2)), int(digits.group(3))
+
+def getLatestTags(tags, num_versions):
+    numericaltags = transformTagsToNumbers(tags)
+    _,y,_ = splitVersion(makeImageTag())
+    latestversions = []
+    ycurr = y
+    for tags in numericaltags[::-1]:
+        if tags[1] < y-num_versions+1:
+            break
+        if tags[1] < ycurr:
+            ycurr = tags[1]
+        if tags[1] == ycurr:
+            latestversions.append(str(tags[0]) + "." + str(tags[1]) + "." + str(tags[2]))
+            ycurr-=1
+    return latestversions
+
+def makeImageTag():
+    return subprocess.check_output(["make", "tag"]).decode(encoding="utf-8")
```

Comment:
Whenever you call `make something` to get its output, as a best practice please apply `--quiet` and `--no-print-directory`. See discussion here https://github.com/stackrox/stackrox/pull/3051#discussion_r968840394
```suggestion
    return subprocess.check_output(["make", "--quiet", "--no-print-directory", "tag"]).decode(encoding="utf-8")
```

Reply:
Good idea, I was worrying about possible side effects from calling `make tag` and had asked about them before, this should help cover edge cases.